exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/interpreter/InterpreterEvaluator.java

```diff
@@ -378,8 +388,14 @@ public ValueHolder visitDoubleConstant(ValueExpressions.DoubleExpression dExpr,
     }
 
     @Override
-    public ValueHolder visitQuotedStringConstant(ValueExpressions.QuotedString e, Integer value) throws RuntimeException {
-      return ValueHolderHelper.getVarCharHolder(getManagedBufferIfAvailable(), e.value);
+    public ValueHolder visitQuotedStringConstant(final ValueExpressions.QuotedString e, Integer value) throws RuntimeException {
+      return getConstantValueHolder(e.value, new Function<DrillBuf, ValueHolder>() {
```

Comment:
Is this constant value cache supposedly to be type-aware?  For example, if I have a QuotedString "123456", and a decimal value 123456,  the cache should return a VarcharHolder for the former, and DecimalHolder for the latter. However, the code seems to produce the same keys for both cases, and may mess up. Or do we have a way to differentiate those two different types?